receiver/pluginreceiver/supplied_plugins_test.go

```diff
@@ -33,6 +33,7 @@ func TestValidateSuppliedPlugins(t *testing.T) {
 	require.NoError(t, err)
 
 	for _, entry := range entries {
+		entry := entry
```

Comment:
is this on purpose? shadowing a variable with a copy of itself?

Reply:
Yes, this is on purpose. I've seen this pattern a few times (e.g. in google's ops agent), it's to fix this issue:

https://stackoverflow.com/questions/69984933/goroutine-inside-for-loop-takes-only-the-last-value-of-slice

Reply:
Oh I see because of the `t.Parrallel`. Yeah I was wondering if you were trying to solve the iterator pointer problem but I didn't see any concurrency at first.

Can we just grab the entry name into a variable since that seems all we want, or rename the variable being assigned to so we don't shadow. I know it should be ok but I'd rather try to avoid confusion where possible.

Reply:
Sure, we can do that. The pattern actually comes from the golang FAQ, btw: 

https://go.dev/doc/faq#closures_and_goroutines